3-extra/2-array-of-objects.js

```diff
@@ -11,7 +11,35 @@
 */
 
 function getHighestRatedInEachGenre(books) {
-    // TODO
+    let maxChil = 0;
+    let nameChil = ''
+    let myarr = []
+    let maxNon = 0;
+    let nameNon = ''
+    let maxCook = 0;
+    let nameCook = ''
+    for (const i in books){
+        if (books[i].rating > maxChil && books[i].genre=="children"){
+            maxChil = books[i].rating
+            nameChil = books[i].title
+        }
+        
+    }
+    for (const i in books){
+        if (books[i].rating > maxNon && books[i].genre=="non-fiction"){
+            maxNon = books[i].rating
+            nameNon = books[i].title
+        }
+        
+    }
+    for (const i in books){
+        if (books[i].rating > maxCook  && books[i].genre=="cooking"){
+            maxCook  = books[i].rating
+            nameCook = books[i].title
+        }
+    
+    }
+    return [nameChil, nameNon, nameCook]
```

Comment:
you can execute in one loop, in this way you will go from a 3xO(n) notation to O(n) notation [speed](https://robbell.io/2009/06/a-beginners-guide-to-big-o-notation) as speed and performance.